src/Azure/Shared/CosmosDB/AzureCosmosDBOptions.cs

```diff
@@ -0,0 +1,116 @@
+using System.Text.Json;
+using Azure.Core;
+
+#if ORLEANS_CLUSTERING
+namespace Orleans.Clustering.CosmosDB;
+#elif ORLEANS_PERSISTENCE
+namespace Orleans.Persistence.CosmosDB;
+#elif ORLEANS_REMINDERS
+namespace Orleans.Reminders.CosmosDB;
+#elif ORLEANS_STREAMING
+namespace Orleans.Streaming.CosmosDB;
+#elif ORLEANS_DIRECTORY
+namespace Orleans.GrainDirectory.CosmosDB;
+#else
+// No default namespace intentionally to cause compile errors if something is not defined
+#endif
+
+public abstract class AzureCosmosDBOptions
+{
+    private const string ORLEANS_DB = "Orleans";
+    private const int ORLEANS_DEFAULT_RU_THROUGHPUT = 400;
+
+    /// <summary>
+    /// Gets or sets the cosmos account endpoint URI. This can be retrieved from the Overview section of the Azure Portal.
+    /// This is required if you are authenticating using tokens.
+    /// <remarks>
+    /// In the form of https://{databaseaccount}.documents.azure.com:443/, see: https://docs.microsoft.com/en-us/rest/api/cosmos-db/cosmosdb-resource-uri-syntax-for-rest
+    /// </remarks>
+    /// </summary>
+    public string AccountEndpoint { get; set; } = default!;
+
+    /// <summary>
+    /// The account key used for a cosmos DB account.
+    /// </summary>
+    [Redact]
+    public string? AccountKey { get; set; }
+
+    /// <summary>
+    /// Tries to create the database and container used for clustering if it does not exist.
+    /// </summary>
+    public bool CanCreateResources { get; set; }
```

Comment:
```suggestion
    public bool IsCosmosResouceCreationEnabled { get; set; }
```

Maybe this name might read a little better?

Reply:
Both seem to align with the .NET Framework Design Guidelines https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-properties

"IsX" is probably better, since the "Can" examples there are about capability of an object, rather than permission.

Including product names in property names feels a bit awkward to me, so I would likely omit "Cosmos". It ought to be apparent by the context. If it's not apparent, then perhaps "Resource" can be changed to something else, or it could be DatabaseResource. Unsure. We could also take a follow-up PR to clean some of it up

Should this default to true? Azure Storage providers do create resources automatically, but ADO.NET ones do not (since they need scripts to be configured)

Reply:
I'm always against automatic operations that modify structure or resources like EFCore migrations for example. It should be explicit that you want to allow that, so I would suggest the default to be `false`. 

In the naming, I would agree with @ReubenBond and would suggest `IsResourceCreationEnabled`.

Thoughts?

Reply:
I like that option @galvesribeiro

Reply:
Done